storage: etcd: use sub-tests

stevekuznetsov opened the pull request:
Sub-tests are much friendlier to anyone breaking these tests than the old-school methods.

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/kind cleanup
/sig api-machinery
/cc @stts @liggitt 

```release-note
NONE
```

```docs

```

stevekuznetsov committed:
storage: etcd: TestGetToList: use sub-tests

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

stevekuznetsov committed:
storage: etcd: TestGuaranteedUpdate: use sub-tests

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

stevekuznetsov committed:
storage: etcd: TestUnconditionalDelete: use sub-tests

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

stevekuznetsov committed:
storage: etcd: TestConditionalDelete: use sub-tests

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

liggitt commented on staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:
```diff
@@ -300,34 +300,39 @@ func TestUnconditionalDelete(t *testing.T) {
-			t.Errorf("#%d: pod want=%#v, get=%#v", i, tt.expectedObj, out)
-		}
+			if err != nil {
+				t.Fatalf("#%d: %s: Delete failed: %v", i, tt.name, err)
```
no need for `i` inside a t.Run

stevekuznetsov replied:
ACK - I was looking to make the smallest change, but I can certainly remove the index prefix from these log lines.

liggitt commented on staging/src/k8s.io/apiserver/pkg/storage/etcd3/store_test.go:
```diff
@@ -300,34 +300,39 @@ func TestUnconditionalDelete(t *testing.T) {
 		expectNotFoundErr: false,
-	}, { // test unconditional delete on non-existing key
+	}, {
+		name: "test unconditional delete on non-existing key",
```
names passed to `t.Run` show up in test grid and explode the left-hand side width, right? might need to trim verbosity a little/a lot

stevekuznetsov replied:
@liggitt I'm not extremely familiar with TestGrid rendering, sorry - is there a guideline or length limit I should target? I can take a stab at shortening the names.

liggitt replied:
https://testgrid.k8s.io/presubmits-kubernetes-blocking#pull-kubernetes-unit&include-filter-by-regex=k8s.io%2Fapiserver%2Fpkg%2Fstorage%2Fetcd3 is the ones we currently have

looks like we already have some ~verbose ones, so maybe this one is fine, but some of the ones below that tell little stories should be trimmed ("test GetToList on existing key with minimum resource version set to current resource version, match=minimum")

stevekuznetsov replied:
Yeah, looks like the same comment -> name approach was taken in e.g. `TestList` conversion. I'll add a commit to shorten some of these where I can.

stevekuznetsov committed:
storage: etcd: shorten sub-test names

Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

stevekuznetsov commented:
@liggitt PTAL - was able to shorten them all, removed redundant prefixes and changed stories to explicit things (e.g. "minimum resource version set to current resource version" became "resourceVersion=current")

liggitt commented:
/lgtm
/approve

stevekuznetsov commented:
@liggitt sorry, looks like my editor was being goofy with `gofmt`-on-save, needs another LGTM please

liggitt commented:
/lgtm

fedebongio commented:
/triage accepted
